consult-xref.el

```diff
@@ -0,0 +1,103 @@
+;;; consult-xref.el --- Provides the command `consult-xref' -*- lexical-binding: t; -*-
+
+;; This file is not part of GNU Emacs.
+
+;; This program is free software: you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; Provide support for showing xref with consult, with preview.
+;; Use the function (setup-consult-xref) to set it up.
+
+;;; Code:
+
+(require 'consult)
+(require 'xref)
+
+(defun xref-show-definitions-consult (fetcher alist)
+  "Let the user choose the target definition with completion.
+
+When there is more than one definition, let the user choose
+between them by typing in the minibuffer with completion."
+  (let* ((xrefs (funcall fetcher))
+         (xref-alist (xref--analyze xrefs))
+         xref-alist-with-line-info
+         xref
+         (group-prefix-length
+          ;; FIXME: Groups are not always file names, but they often
+          ;; are.  At least this shouldn't make the other kinds of
+          ;; groups look worse.
+          (let ((common-prefix (try-completion "" xref-alist)))
```

Comment:
Why do we want to determine the prefix using try-completion? Why is it necessary to mark the prefix with a special face?

Reply:
I think you misread the substring call below, the prefix isn't highlighted, but chopped off. The rest of of the file name is given a special face.

Reply:
Right, I still think it would be better to not use try-completion to find the common prefix.